README.md

```diff
@@ -11,17 +11,54 @@ can get their hello.
 First of all switch to your desired namespace. Then follow these instructions
 to have your `hello-server` up and running.
 
-1. Create Deployment
+The `hello-server` application requires a `config` to work. Possible options are providing `config` through a `config.yaml` file, environment variables, or default config.
+Passing config files and environment variables is through a `ConfigMap`. To test each way, comment out the other one's way of loading in `k1s-deployment.yaml` 
 
-```sh
-kubectl apply -f k1-deployment.yaml
-kubectl get deployment
-kubectl get pod
-```
 
-2. Create Service
 
-```sh
-kubectl apply -f k1-svc.yaml
-kubectl get svc
-```
+1. Create ConfigMap
+
+    ```sh
+    kubectl apply -f k1s-config-map.yaml
+    kubectl get configmaps
+    ```
+
+    By default, both `config.yaml` and environment variables are set and `hello-server` prioritizes env variables against config file. For only loading `config.yaml`, delete the `envFrom:` part from `k1s-deployment.yaml`.
+
+2. Create Deployment
+
+    ```sh
+    kubectl apply -f k1s-deployment.yaml
+    kubectl get deployment
+    kubectl get pod
+    ```
+
+    To visualize the usage of `health-check` you can use `/die` endpoint to make `health-check` fail. and then check what happened with `kubectl get events`.
+
+3. Create Service
+
+    ```sh
+    kubectl apply -f k1s-svc.yaml
+    kubectl get svc
+    ```
+
+    `k1s` service is used to associate a name for `k1s-deployment` pod's IP addresses inside cluster. You can create a shell session inside one of pods from `kubectl get pods` and then try  :
```

Comment:
I think it would be better if you remove the space in `try :` and write `try:`

Reply:
done.